geopandas/geoseries.py

```diff
@@ -512,7 +526,7 @@ def _from_wkb_or_wkb(
         return cls(from_wkb_or_wkt_function(data, crs=crs), index=index, **kwargs)
 
     @property
-    def __geo_interface__(self):
+    def __geo_interface__(self) -> dict:
```

Comment:
```suggestion
    def __geo_interface__(self) -> Dict:
```
Using lowercase dict is a python 3.9 feature, so technically, we should use the one from typing for now
https://docs.python.org/3/whatsnew/3.9.html#type-hinting-generics-in-standard-collections